### PR TITLE
Fix tensorflow requirements for ARM Macs

### PR DESCRIPTION
**Pull request by astrojuanlu**
Fix gh-141.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by noklam** [comment]
@ankatiyar Can we look into adding Mac or Arm? I know GitHub action support Mac for sure, we may not need to pass all checking for this hardware but it will be nice to know if we can support it.
* https://github.com/kedro-org/kedro/issues/2153

File: kedro-datasets/setup.py [hunk 1]
```diff
@@ -86,7 +86,8 @@ def _collect_requirements(requires):
     "tensorflow.TensorflowModelDataset": [
         # currently only TensorFlow V2 supported for saving and loading.
         # V1 requires HDF5 and serialises differently
-        "tensorflow~=2.0"
+        "tensorflow-macos~=2.0; platform_system == 'Darwin' and platform_machine == 'arm64'",
+        "tensorflow~=2.0; platform_system != 'Darwin' or platform_machine != 'arm64'",
```
Comment (noklam) [inline kedro-datasets/setup.py:90]: https://www.tensorflow.org/install/pip#macos_1

The official guide for MacOS is `pip install tensorflow`? Can you supplement the link where it suggest `tensorflow-macos`?

**Review by noklam — COMMENTED** [review]
It reminds me the old day when `theano` is still the backend and installing tensorflow and CUDA properly is harder than the deep learning part🤣.

File: kedro-datasets/setup.py (hunk 1 above)
Comment (astrojuanlu) [reply kedro-datasets/setup.py:90]: Yep:

> For users of Apple M1 computers, to get native performance, you'll want to follow the instructions found [here](https://developer.apple.com/metal/tensorflow-plugin/).

image

So it should be this guide https://developer.apple.com/metal/tensorflow-plugin/ I'll add a comment

**Review by noklam — APPROVED** [review]
It looks fine for me but would be great to have someone have a M1 machine to try this out. Maybe @stichbury ?

**Comment by astrojuanlu** [comment]
I tried this locally on my M1 Mac and it works, but I agree it would be nice to have someone else try it just in case 😄

**Comment by deepyaman** [comment]
@astrojuanlu Wait, you mean commenting out the `tensorflow` requirement is not the best way to develop? :joy: TIL.

I'll try this out, thanks!

**Comment by astrojuanlu** [comment]
> Wait, you mean commenting out the tensorflow requirement is not the best way to develop? 😂 

What are you talking about? I've never done such a thing... 😂

**Pull request merged**